satellite/satellitedb/overlaycache.go

```diff
@@ -1338,3 +1338,126 @@ func populateUpdateFields(dbNode *dbx.Node, updateReq *overlay.UpdateRequest) db
 
 	return updateFields
 }
+
+// UpdateCheckIn updates a single storagenode with info from when the the node last checked in.
+func (cache *overlaycache) UpdateCheckIn(ctx context.Context, node overlay.NodeCheckInInfo, config overlay.NodeSelectionConfig) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	if node.Address.GetAddress() == "" {
+		return Error.New("error UpdateCheckIn: missing the storage node address")
+	}
+
+	switch t := cache.db.Driver().(type) {
+	case *sqlite3.SQLiteDriver:
+		value := pb.Node{
+			Id: node.NodeID,
+			Address: &pb.NodeAddress{
+				Transport: pb.NodeTransport_TCP_TLS_GRPC,
+				Address:   node.Address.GetAddress(),
+			},
+		}
+		err := cache.UpdateAddress(ctx, &value, config)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+
+		_, err = cache.UpdateUptime(ctx, node.NodeID, node.IsUp, config.UptimeReputationLambda, config.UptimeReputationWeight, config.UptimeReputationDQ)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+
+		pbInfo := pb.InfoResponse{
+			Operator: node.Operator,
+			Capacity: node.Capacity,
+			Type:     pb.NodeType_STORAGE,
+		}
+		_, err = cache.UpdateNodeInfo(ctx, node.NodeID, &pbInfo)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+	case *pq.Driver:
+		// v is a single feedback value that allows us to update both alpha and beta
+		var v float64 = -1
+		uptimeSuccess := 0
+		lastContactSuccess := time.Time{}.UTC()
+		lastContactFailure := time.Time{}.UTC()
+		if node.IsUp {
+			v = 1
+			uptimeSuccess = 1
+			lastContactSuccess = time.Now().UTC()
+			lastContactFailure = time.Time{}.UTC()
+		}
+		uptimeReputationAlpha := config.UptimeReputationLambda*config.UptimeReputationAlpha0 + config.UptimeReputationWeight*(1+v)/2
+		uptimeReputationBeta := config.UptimeReputationLambda*config.UptimeReputationBeta0 + config.UptimeReputationWeight*(1-v)/2
+		start := time.Now()
+		query := `
+			INSERT INTO nodes
+			(
+				id, address, last_net, protocol, type,
+				email, wallet, free_bandwidth, free_disk,
+				audit_success_count, total_audit_count, uptime_success_count, total_uptime_count,
+				created_at, updated_at, last_contact_success, last_contact_failure,
+				audit_reputation_alpha, audit_reputation_beta, uptime_reputation_alpha, uptime_reputation_beta,
+				contained, disqualified,
+				piece_count, major, minor, patch, hash, timestamp, release, latency_90
+			)
+			VALUES (
+				$1, $2, $3, $4, $5,
+				$6, $7, $8, $9,
+				0, 0, $10, 1,
+				current_timestamp, current_timestamp, $11, $12,
+				$21, $22, $13, $14,
+				false, null,
+				0, 0, 0, 0, '', $15, false, 0
+			)
+			ON CONFLICT (id)
+			DO UPDATE
+			SET
+				total_uptime_count=nodes.total_uptime_count+1,
+				uptime_reputation_alpha=$18::numeric*nodes.uptime_reputation_alpha + $19::numeric*(1+$20)/2,
```

Comment:
The way it is now, `v` aka `$20` can be either `1` or `-1`.
Replacing the variable `$20` with `$16::bool::int` changes the option of `1`/`-1` to `1`/`0`.  I'm not sure this would work for the negative case/

Reply:
Oh yeah, don't replace `$20` with `$16::bool::int` directly. You'd have to replace the whole expression `(1+$20)/2` with `$16::bool::int` for this to work. 

    (1+1)/2 = 1
    (-1+1)/2 = 0

Reply:
ohhhh ok i see what you mean, let me try that out.